File: src/main/web/src/app/service/review.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {Observable} from "rxjs";
import {Restaurant} from "../models/restaurant.model";
import {Review} from "../models/review.model";

@Injectable({
  providedIn: 'root'
})
export class ReviewService {
  resourceUrl = '/review';

  constructor(private http: HttpClient) { }

  addReview(restaurantId: number, review: string): Observable<Review> {
    return this.http.post<Review>(`${this.resourceUrl}/${restaurantId}`, review);
  }

  getAllReviews(restaurantId: number): Observable<Review[]> {
    return this.http.get<Review[]>(`${this.resourceUrl}/${restaurantId}`);
  }

  deleteReview(id: number) {
    return this.http.delete(`${this.resourceUrl}/${id}`)
  }

}

File: src/main/java/com/tvz/foodfinder/service/ReviewService.java
package com.tvz.foodfinder.service;

import com.tvz.foodfinder.domain.Review;

import java.util.List;

public interface ReviewService {

    List<Review> getAllReviews(Long restaurantId);

    Review addReview(Long restaurantId, String review);

    void deleteReview(Long id);

}
